cvat-ui/src/components/accept-invitation-page/accept-invitation-page.tsx

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2023 CVAT.ai Corporation
+//
+// SPDX-License-Identifier: MIT
+
+import React, { useCallback } from 'react';
+import RegisterPageComponent from 'components/register-page/register-page';
+import { CombinedState } from 'reducers';
+import { useSelector, useDispatch } from 'react-redux';
+import { useHistory } from 'react-router';
+import { acceptInvitationAsync } from 'actions/auth-actions';
+import { RegisterData } from 'components/register-page/register-form';
+
+interface InvitationParams {
+    email: string;
+    key: string;
+}
+
+function AcceptInvitationPage(): JSX.Element {
+    const userAgreements = useSelector((state: CombinedState) => state.userAgreements.list);
+    const userAgreementsFetching = useSelector((state: CombinedState) => state.userAgreements.fetching);
+    const authFetching = useSelector((state: CombinedState) => state.auth.fetching);
+    const history = useHistory();
+    const dispatch = useDispatch();
+    const queryParams = new URLSearchParams(history.location.search);
+    const invitationParams: InvitationParams = {
+        email: queryParams.get('email') || '',
+        key: queryParams.get('key') || '',
+    };
+    const onRegister = useCallback((registerData: RegisterData) => {
+        dispatch(acceptInvitationAsync(
+            registerData,
+            invitationParams.key,
+            (orgSlug: string) => {
+                history.replace(`/auth/login?next=/tasks&activateOrganization=${orgSlug}`);
+            },
```

Comment:
I do not think approach with activate command in query is good enough. Why do we need this? Why can't we activate organization in this callback? Why can't we add org slug to local storage and reload the page? Even not sure that we need to reload, we can look at current org in org watcher where now query parameter is tracked in onmount hook

Reply:
I've checked it, we can set `currentOrganization` and it will be automatically activated without reload, just on login